saleor/plugins/webhook/tasks.py

```diff
@@ -1,910 +0,0 @@
-                message,
-                saleorDomain=domain,
-                saleorApiUrl=build_absolute_uri(reverse("api"), domain),
-                eventType=event_type,
-                signature=signature,
-            )
-        except (pubsub_v1.publisher.exceptions.MessageTooLargeError, RuntimeError) as e:
-            return WebhookResponse(content=str(e), status=EventDeliveryStatus.FAILED)
-        response_duration = duration()
-        response = future.result()
-        return WebhookResponse(content=response, duration=response_duration)
-
-
-def send_webhook_using_scheme_method(
-    target_url,
-    domain,
-    secret,
-    event_type,
-    data,
-    custom_headers=None,
-) -> WebhookResponse:
-    parts = urlparse(target_url)
-    message = data.encode("utf-8")
-    signature = signature_for_payload(message, secret)
-    scheme_matrix: Dict[WebhookSchemes, Callable] = {
-        WebhookSchemes.HTTP: send_webhook_using_http,
-        WebhookSchemes.HTTPS: send_webhook_using_http,
-        WebhookSchemes.AWS_SQS: send_webhook_using_aws_sqs,
-        WebhookSchemes.GOOGLE_CLOUD_PUBSUB: send_webhook_using_google_cloud_pubsub,
-    }
-
-    if send_method := scheme_matrix.get(parts.scheme.lower()):
-        # try:
-        return send_method(
-            target_url,
-            message,
-            domain,
-            signature,
-            event_type,
-            custom_headers=custom_headers,
-        )
-    raise ValueError("Unknown webhook scheme: %r" % (parts.scheme,))
-
-
-def handle_webhook_retry(
-    celery_task, webhook, response_content, delivery, delivery_attempt
-) -> bool:
-    """Handle celery retry for webhook requests.
-
-    Calls retry to re-run the celery_task by raising Retry exception.
-    When MaxRetriesExceededError is raised the function will end without exception.
-    """
-    is_success = True
-    task_logger.info(
-        "[Webhook ID: %r] Failed request to %r: %r for event: %r."
-        " Delivery attempt id: %r",
-        webhook.id,
-        webhook.target_url,
-        response_content,
-        delivery.event_type,
-        delivery_attempt.id,
-    )
-    try:
-        countdown = celery_task.retry_backoff * (2**celery_task.request.retries)
-        celery_task.retry(countdown=countdown, **celery_task.retry_kwargs)
-    except Retry as retry_error:
-        next_retry = observability.task_next_retry_date(retry_error)
-        observability.report_event_delivery_attempt(delivery_attempt, next_retry)
-        raise retry_error
-    except MaxRetriesExceededError:
-        is_success = False
-        task_logger.info(
-            "[Webhook ID: %r] Failed request to %r: exceeded retry limit."
-            "Delivery id: %r",
-            webhook.id,
-            webhook.target_url,
-            delivery.id,
-        )
-    return is_success
-
-
-@app.task(
-    queue=settings.WEBHOOK_CELERY_QUEUE_NAME,
-    bind=True,
-    retry_backoff=10,
-    retry_kwargs={"max_retries": 5},
-)
```

Comment:
There is something we didn't discuss before - we need to keep Celery tasks in their original paths for backward compatibility. Ideally, we should just keep this file with the necessary Celery tasks, and if possible import logic from the new location.

If we removed these tasks it could break during deployment - if there were any unfinished tasks in the queue, and the new code was deployed, Celery would fail as it wouldn't be able to locate these tasks. If this PR is released in 3.17, we will be able to remove these tasks in Saleor 3.18.